### PR TITLE
Add support for printing SCTs

### PR DESCRIPTION
**Pull request by jdtw**
Fixes https://github.com/square/certigo/issues/171

Example from `certigo connect -v google.com`:

```
Signed Certificate Timestamp:
	Version: 0 (v1)
	Log Operator: Google
	Log URL: https://ct.googleapis.com/logs/xenon2022/
	Log ID: 46:A5:55:EB:75:FA:91:20:30:B5:A2:89:69:F4:F3:7D:
	        11:2C:41:74:BE:FD:49:B8:85:AB:F2:FC:70:FE:6D:47
	Timestamp: 2022-06-06 02:29 PDT
	Signature: ECDSA-SHA256
Signed Certificate Timestamp:
	Version: 0 (v1)
	Log Operator: DigiCert
	Log URL: https://nessie2022.ct.digicert.com/log/
	Log ID: 51:A3:B0:F5:FD:01:79:9C:56:6D:B8:37:78:8F:0C:A4:
	        7A:CC:1B:27:CB:F7:9E:88:42:9A:0D:FE:D4:8B:05:E5
	Timestamp: 2022-06-06 02:29 PDT
	Signature: ECDSA-SHA256
```

We keep a hard-coded list of known logs that can be updated by running `make ctlogs` from the repo root.

### CODE REVIEW

File: lib/ct.go [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+}
+
+func getLogByID(id []byte) *ctLog {
+	knownLogsOnce.Do(func() {
+		client := &http.Client{
+			// Set a timeout so we don't block forever on broken servers.
+			Timeout: 5 * time.Second,
+		}
+
+		resp, err := client.Get(knownLogsAddr)
```
Comment (jdtw) [inline lib/ct.go:74]: Making a network call to get the current list of known logs during, say, `certigo dump` is less than ideal (but will only happen if the cert has SCTs). Another alternative is to hardcode a list of known logs, but I'm not sure we should sign up to maintain that. And finally, a third option would be to only fetch the log URL and operator if an explicit flag was passed, but that would require a fair bit of plumbing to accomplish (but is probably the right thing to do?)

File: lib/display.go [hunk 2]
```diff
@@ -156,34 +170,6 @@ Email Addresses:
 Warnings:{{range .Warnings}}
 	{{. | redify}}{{end}}{{end}}`
 
-type certWithName struct {
```
Comment (jdtw) [inline lib/display.go:159]: These are all unused.

File: lib/ct.go (hunk 1 above)
Comment (jdtw) [reply lib/ct.go]: I decided to keep a hardcoded list to avoid the network call at runtime.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,5 @@
+.PHONY: ctlogs
```
Comment (sq-smits) [inline Makefile:1]: Would it make sense to move this Makefile into the gen-known-logs directory? The issue being that people might assume that "make" should build certigo, which is false.

File: internal/gen-known-logs/main.go [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+			} `json:"logs"`
+		} `json:"operators"`
+	}
+
+	if err := json.NewDecoder(resp.Body).Decode(&logs); err != nil {
+		log.Printf("Failed to parse the list of known CT logs: %v", err)
+		return
+	}
+
+	fmt.Printf(`// Autogenerated with github.com/square/certigo/internal/gen-known-logs
```
Comment (sq-smits) [inline internal/gen-known-logs/main.go:46]: Would it make sense to include a timestamp on when it was last generated?

**Review by sq-smits — APPROVED** [review]

File: internal/gen-known-logs/Makefile [hunk 5]
```diff
@@ -0,0 +1,5 @@
+.PHONY: ctlogs
+
+ctlogs:
+	go run github.com/square/certigo/internal/gen-known-logs --package lib > ../../lib/ctlogs.go
```
Comment (mbyczkowski) [inline internal/gen-known-logs/Makefile:4]: Would this be a good candidate for `go:generate`? https://pkg.go.dev/cmd/go#hdr-Generate_Go_files_by_processing_source

Comment (jdtw) [reply internal/gen-known-logs/Makefile:4]: Yes, good call. https://github.com/square/certigo/pull/279

**Pull request merged**